lib/ocpp/v16/charge_point_impl.cpp

```diff
@@ -2737,6 +2870,248 @@ void ChargePointImpl::handleGetLocalListVersionRequest(ocpp::Call<GetLocalListVe
+            if (triggerMeterValue.is_object()) {
+                if (triggerMeterValue.contains("atTime")) {
+                    trigger_meter_value_at_time = triggerMeterValue.at("atTime");
+                }
+
+                if (triggerMeterValue.contains("atEnergykWh")) {
+                    trigger_meter_value_at_energy_kwh = triggerMeterValue.at("atEnergykWh");
+                }
+
+                if (triggerMeterValue.contains("atPowerkW")) {
+                    trigger_meter_value_at_power_kw = triggerMeterValue.at("atPowerkW");
+                }
+
+                if (triggerMeterValue.contains("atCPStatus")) {
+                    std::vector<ChargePointStatus> trigger_cp_status;
+                    json array;
+                    for (const auto& cp_status : triggerMeterValue.at("atCPStatus").items()) {
+                        try {
+                            trigger_cp_status.push_back(
+                                v16::conversions::string_to_charge_point_status(cp_status.value().get<std::string>()));
+                        } catch (const std::out_of_range& e) {
+                            EVLOG_error << "Could not trigger on CP status: status ("
+                                        << cp_status.value().get<std::string>()
+                                        << ") is not a valid chargepoint status: " << e.what();
+                        }
+                    }
+                    if (trigger_cp_status.size() > 0) {
+                        trigger_meter_value_at_chargepoint_status = trigger_cp_status;
+                    }
+                }
+            }
+        }
+    } catch (const std::exception& e) {
+        EVLOG_warning << "Set session cost: Parsing failed! " << e.what();
+        response.data = "json parsing failed";
+        return response;
+    }
+
+    int32_t connector_id = this->transaction_handler->get_connector_from_transaction_id(std::stoi(cost.transaction_id));
+    std::shared_ptr<Connector> connector = this->connectors.at(connector_id);
+    std::string session_id = cost.transaction_id;
+    if (connector_id == -1) {
+        EVLOG_warning << "Set session cost: Could not set session id because transaction with transaction id "
+                      << cost.transaction_id << " is not found. Using transaction id as session id.";
+    } else {
+        std::shared_ptr<Transaction> transaction = this->transaction_handler->get_transaction(connector_id);
+        if (transaction != nullptr) {
+            session_id = transaction->get_session_id();
+        } else {
+            EVLOG_warning << "Set session cost: Could not set session id because transaction with transaction id "
+                          << cost.transaction_id << " for connector " << connector_id
+                          << " is not found. Using transaction id as session id.";
+        }
+    }
+
+    cost.transaction_id = session_id;
+
+    if (connector != nullptr) {
```

Comment:
I think it's not is it? Because there is no session so we can not show anything to the user. 

Actually same as this: https://github.com/EVerest/libocpp/pull/707#discussion_r1695273557


So I think if there is no session id found: return rejected and not continue (in both cases)?